packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -405,14 +399,15 @@ export function getFocusableTreeWalker(root: HTMLElement, opts?: FocusManagerOpt
           return NodeFilter.FILTER_REJECT;
         }
 
-        if ((node as HTMLElement).matches(selector)) {
+        if ((node as HTMLElement).matches(selector)
+          && isElementVisible(node as HTMLElement)
+          && (!scope || isElementInScope(node as HTMLElement, scope))) {
           return NodeFilter.FILTER_ACCEPT;
         }
 
         return NodeFilter.FILTER_SKIP;
       }
-    },
-    false
```

Comment:
I know it's not quite your business as you don't support IE11 but I have just wanted to leave this information here for future reference. Removing this argument has broken compatibility with IE11 as it doesn't treat it as optional.